### DISCUSSION TITLE
Fetching data from database view?

### DISCUSSION THREAD

**Discussion by Jasenkoo** [mikro-orm/mikro-orm#2872] [Q&A] [answered]
Is there any way we can fetch data from the database view?

In typeorm I know there is an option where you can define an entity with `@ViewEntity`, so I was wondering is there something similar in MikroORM?

I tried using `@Entity` but got an error on connection for missing `@PrimaryKey`

**Comment by B4nan** [comment, marked as answer]
Nope, related issues: #672 and #1104

I'd like to do the #1104 approach first, to allow having "expression entity" which should help with both, the #672 is also about handing real db views and schema diffing on them.

**Reply by Jasenkoo** [reply]
Tnx for the answer, checked the linked issues. Ended up creating a raw query then mapping it to a class.
